nex/account-management/nintendo_create_account.go

```diff
@@ -8,64 +8,76 @@ import (
 	"encoding/hex"
 	"strings"
 
-	"github.com/PretendoNetwork/friends-secure/globals"
-	"github.com/PretendoNetwork/friends-secure/utility"
+	"github.com/PretendoNetwork/friends/globals"
+	"github.com/PretendoNetwork/friends/utility"
 	nex "github.com/PretendoNetwork/nex-go"
 	account_management "github.com/PretendoNetwork/nex-protocols-go/account-management"
 	account_management_types "github.com/PretendoNetwork/nex-protocols-go/account-management/types"
 )
 
-func NintendoCreateAccount(err error, client *nex.Client, callID uint32, strPrincipalName string, strKey string, uiGroups uint32, strEmail string, oAuthData *nex.DataHolder) {
+func NintendoCreateAccount(err error, client *nex.Client, callID uint32, strPrincipalName string, strKey string, uiGroups uint32, strEmail string, oAuthData *nex.DataHolder) uint32 {
 	if err != nil {
-		// TODO: Handle error
-		globals.Logger.Critical(err.Error())
+		globals.Logger.Error(err.Error())
+		return nex.Errors.Core.InvalidArgument
 	}
 
-	rmcResponse := nex.NewRMCResponse(account_management.ProtocolID, callID)
-
 	var tokenBase64 string
 
 	oAuthDataType := oAuthData.TypeName()
 
-	if oAuthDataType == "NintendoCreateAccountData" { // Wii U
+	switch oAuthDataType {
+	case "NintendoCreateAccountData": // Wii U
 		nintendoCreateAccountData := oAuthData.ObjectData().(*account_management_types.NintendoCreateAccountData)
 
 		tokenBase64 = nintendoCreateAccountData.Token
-	} else if oAuthDataType == "AccountExtraInfo" { // 3DS
+	case "AccountExtraInfo": // 3DS
 		accountExtraInfo := oAuthData.ObjectData().(*account_management_types.AccountExtraInfo)
 
 		tokenBase64 = accountExtraInfo.NEXToken
 		tokenBase64 = strings.Replace(tokenBase64, ".", "+", -1)
 		tokenBase64 = strings.Replace(tokenBase64, "-", "/", -1)
 		tokenBase64 = strings.Replace(tokenBase64, "*", "=", -1)
+	default:
+		globals.Logger.Errorf("Invalid oAuthData data type %s!", oAuthDataType)
+		return nex.Errors.Authentication.TokenParseError

```

Comment:
Is this the real error which is sent in these cases?

Reply:
Calling this method with dummy data, including replacing the oAuthData with friends.Mii raises `PythonCore.Exception`, although I think we are fine by using this error code

Reply:
Ideally I would like to be ass accurate as possible, even when it comes to errors

Though that error sounds...different. It sounds like the real servers aren't actually checking the DataHolder type, which leads to an exception? So maybe you're right that we should use a different error, because that's not *actually* what's happening here for us, there is no exception

Reply:
Should I keep this error then?

Reply:
Yeah this is fine unless we find a better one in the future